proxy/src/bin/pg_sni_router.rs

```diff
@@ -109,20 +110,25 @@ async fn main() -> anyhow::Result<()> {
 
     let cancellation_token = CancellationToken::new();
 
-    let main = proxy::flatten_err(tokio::spawn(task_main(
+    let main = tokio::spawn(task_main(
         Arc::new(destination),
         tls_config,
         proxy_listener,
         cancellation_token.clone(),
-    )));
-    let signals_task = proxy::flatten_err(tokio::spawn(proxy::handle_signals(cancellation_token)));
-
-    tokio::select! {
-        res = main => { res?; },
-        res = signals_task => { res?; },
-    }
+    ));
+    let signals_task = tokio::spawn(proxy::handle_signals(cancellation_token));
+
+    // the signal task cant ever succeed.
+    // the main task can error, or can succeed on cancellation.
+    // we want to immediately exit on either of these cases
+    let signal = match futures::future::select(signals_task, main).await {
```

Comment:
actually I feel tokio::select would be more readable here but I'm fine with using `select` to make it easier for rustfmt and potential cancellation things

Reply:
I have an agenda against `tokio::select!{}` 😂

I personally find the use of future::select much clearer in proxy.rs so while I think you're probably correct that tokio::select looks nicer in this scenario, I wanted to keep it consistent. Open for debate on this though